dtfs-central-api/src/v1/controllers/utilisation-report-service/post-validate-utilisation-report-data.controller/index.test.ts

```diff
@@ -0,0 +1,117 @@
+import httpMocks from 'node-mocks-http';
+import { ApiError } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { PostValidateUtilisationReportDataPayload } from '../../../routes/middleware/payload-validation';
+import { postValidateUtilisationReportData, PostValidateUtilisationReportDataRequest } from '.';
+import { validateUtilisationReportCsvData } from '../../../../services/utilisation-report-data-validator';
+
+jest.mock('../../../../services/utilisation-report-data-validator');
+
+console.error = jest.fn();
+
+class TestApiError extends ApiError {
+  constructor(status?: number, message?: string) {
+    super({ status: status ?? 500, message: message ?? '' });
+  }
+}
+
+describe('post-validate-utilisation-report-data.controller', () => {
+  describe('postValidateUtilisationReportData', () => {
+    const aValidRequestBody = (): PostValidateUtilisationReportDataPayload => ({
+      reportData: [{ header: { value: 'value', column: 'A', row: '1' } }],
+    });
+
+    it('return 200 (Ok) and validation errors when successfully validates the report data', () => {
+      // Arrange
+      const reportData = [{ header: { value: 'value', column: 'A', row: '1' } }];
+      const req = httpMocks.createRequest<PostValidateUtilisationReportDataRequest>({
+        body: { reportData },
+      });
+      const res = httpMocks.createResponse();
+
+      jest.mocked(validateUtilisationReportCsvData).mockReturnValue([{ errorMessage: 'Data invalid!' }]);
+
+      // Act
+      postValidateUtilisationReportData(req, res);
+
+      // Assert
+      expect(validateUtilisationReportCsvData).toHaveBeenCalledWith(reportData);
+      expect(res._getStatusCode()).toBe(HttpStatusCode.Ok);
+      expect(res._getData()).toEqual({ csvValidationErrors: [{ errorMessage: 'Data invalid!' }] });
```

Comment:
It feels so strange to respond with a 200 with validation errors, but I like this solution